### PR TITLE
Fix compiler warnings, define feature toggles, general cleanup

### PR DESCRIPTION
**Pull request by alfreb**
Depends on https://github.com/includeos/IncludeOS/pull/2260 - a full test.sh passes, with the exception of the one unit test remaining to be fixed, `fixed_list_alloc_test_unit`.

### CODE REVIEW

**Review by MagnusS — APPROVED** [review]
LGTM! (see minor comments)

File: api/features.hpp [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ * 5. Add CMake options for toggles as the features mature.
+ *
+ * 6. Put experimental / incomplete features in an 'experimental' namespace to
+ * clearly communicate this to users.
+ */
+
+#ifndef INCLUDEOS_EXPERIMENTAL_MLD2
+#define INCLUDEOS_EXPERIMENTAL_MLD2 false
+#endif
+
```
[the remark this answers is not in the document]
Comment (alfreb) [reply api/features.hpp:34]: Good idea! I'll look into it and do another pass

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -46,7 +46,7 @@ if (DEBUG_INFO)
   set(NO_DEBUG "")
 endif()
 
-set(CMAKE_CXX_FLAGS "-g -O0  -std=c++17 -Wall -Wextra -Wno-unused-function -D__id_t_defined -DUNITTESTS -DURI_THROW_ON_ERROR ${NO_INFO} ${NO_DEBUG} -DGSL_THROW_ON_CONTRACT_VIOLATION -Dlest_FEATURE_AUTO_REGISTER=1 -DHAVE_LEST_MAIN")
+set(CMAKE_CXX_FLAGS "-g -O0 -std=c++17 -Wall -Wextra -Wno-frame-address -Wno-unused-function -Wno-int-to-pointer-cast -D__id_t_defined -DUNITTESTS -DURI_THROW_ON_ERROR ${NO_INFO} ${NO_DEBUG} -DGSL_THROW_ON_CONTRACT_VIOLATION -Dlest_FEATURE_AUTO_REGISTER=1 -DHAVE_LEST_MAIN")
```
[the remark this answers is not in the document]
Comment (alfreb) [reply test/CMakeLists.txt:49]: Good catch, thanks!

File: api/features.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (MagnusS) [reply api/features.hpp:34]: This and #2260 are quite big and almost ready to merge, perhaps we should merge those first? Then additional options could go in a follow up PR later

File: test/CMakeLists.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (alfreb) [reply test/CMakeLists.txt]: Fixed

**Pull request merged**